aws/src/integration/java/org/apache/iceberg/aws/AwsIntegTestUtil.java

```diff
@@ -95,17 +98,16 @@ public static String testAccountId() {
     return System.getenv("AWS_TEST_ACCOUNT_ID");
   }
 
-  public static void cleanS3Bucket(S3Client s3, String bucketName, String prefix) {
+  public static void cleanS3GeneralBucket(S3Client s3, String bucketName, String prefix) {
     ListObjectVersionsIterable response =
         s3.listObjectVersionsPaginator(
             ListObjectVersionsRequest.builder().bucket(bucketName).prefix(prefix).build());
     List<ObjectVersion> versionsToDelete = Lists.newArrayList();
-    int batchDeletionSize = 1000;
     response.versions().stream()
         .forEach(
             version -> {
               versionsToDelete.add(version);
-              if (versionsToDelete.size() == batchDeletionSize) {
+              if (versionsToDelete.size() == BATCH_DELETION_SIZE) {
                 deleteObjectVersions(s3, bucketName, versionsToDelete);
                 versionsToDelete.clear();
               }
```

Comment:
I know this is existing code, but how confident are we that this runs sequentially?

Quick look at `forEach` in JavaDoc says: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#forEach

> The behavior of this operation is explicitly nondeterministic. For parallel stream pipelines, this operation does not guarantee to respect the encounter order of the stream, as doing so would sacrifice the benefit of parallelism

Reply:
I'd always be careful with modifying state inside a stream `map` / `iter` or `forEach`